### PR TITLE
Change to use the PAL TLS API.

### PR DESCRIPTION
**Pull request by simosillankorva**
Needs a small change to PAL TLS before it can work and be merged. Also needs a new function to be added to connectionsecurity.h in mbed-client (I will create the pull request for that).

### CODE REVIEW

**Comment by simosillankorva** [comment]
Pull request done to PAL: https://github.com/ARMmbed/mbed-client-pal/pull/12 .

**Comment by simosillankorva** [comment]
This change to connection handler is also required for this to work: https://github.com/ARMmbed/mbed-client-classic/pull/52 .

File: mbed-client-mbedtls/m2mconnectionsecuritypimpl.h [hunk 1]
```diff
@@ -21,32 +21,25 @@
...
-#include "mbedtls/error.h"
-#include "mbedtls/certs.h"
-#include "mbedtls/entropy_poll.h"
+extern "C"{
+#include "pal_TLS.h"
+}
 
 class M2MTimer;
 
 //TODO: Should we let application to select these or not??
```
Comment (JanneKiiskila) [inline mbed-client-mbedtls/m2mconnectionsecuritypimpl.h]: If we can comment this out and it doesnt' matter, should we not just remove the whole block?

**Comment by yogpan01** [comment]
This PR is still missing the ability to add own entropy for the client because that bit is coming from Crypto implementation. This PR has to also include Crypto API usage from PAL which will be ready after next sprint.

File: source/m2mconnectionsecuritypimpl.cpp [hunk 2]
```diff
@@ -171,82 +178,76 @@ int M2MConnectionSecurityPimpl::connect(M2MConnectionHandler* connHandler){
...
-
-    int ret =- 1;
+int M2MConnectionSecurityPimpl::read(unsigned char* buffer, uint16_t len)
+{
+    int ret = -1;
     palStatus_t return_value;
     uint32_t len_read;
 
-    if(!_init_done){
+    if(2 != _init_done){
```
Comment (JanneKiiskila) [inline source/m2mconnectionsecuritypimpl.cpp]: why this change?

File: source/m2mconnectionsecuritypimpl.cpp [hunk 3]
```diff
@@ -146,22 +152,23 @@ int M2MConnectionSecurityPimpl::start_handshake(){
...
-int M2MConnectionSecurityPimpl::connect(M2MConnectionHandler* connHandler){
-
+int M2MConnectionSecurityPimpl::connect(M2MConnectionHandler* connHandler)
+{
     tr_debug("M2MConnectionSecurityPimpl::connect");
-    int ret=-1;
-    if(!_init_done){
+    int ret = -1;
+
+    if(2 != _init_done){
```
Comment (JanneKiiskila) [inline source/m2mconnectionsecuritypimpl.cpp]: Where is this hard-coded 2 coming from? Needs to be literal or in minimum why its 2? 
it's a much stricter comparision than the previous !_init_done, why?

File: mbed-client-mbedtls/m2mconnectionsecuritypimpl.h [hunk 4]
```diff
@@ -24,7 +24,10 @@
 extern "C"{
 #include "pal_TLS.h"
 }
-
+#ifdef USE_CUSTOM_MBEDTLS_ENTROPY
```
Comment (JanneKiiskila) [inline mbed-client-mbedtls/m2mconnectionsecuritypimpl.h]: The IFDEF should be renamed so that it's matching the rest of the MBED_CLOUD_CLIENT configs AND also the config should be added to the mbed-cloud-client-example config.

Comment (JanneKiiskila) [inline mbed-client-mbedtls/m2mconnectionsecuritypimpl.h]: https://github.com/ARMmbed/mbed-cloud-client-example/blob/master/mbed_cloud_client_user_config.h

**Comment by JanneKiiskila** [comment]
Nitpick - once again commit message with imperative, not imperfect. "Changed" -> "Change" or rather "Take PAL TLS API into use" or similar.

**Comment by simosillankorva** [comment]
There is a new pull request created for this change: https://github.com/ARMmbed/mbed-client-mbed-tls/pull/63.
